src/driver/op.rs

```diff
@@ -83,57 +101,72 @@ impl From<cqueue::Entry> for CqeResult {
     }
 }
 
-impl<T, CqeType> Op<T, CqeType>
-where
-    T: Completable,
-{
+impl<T, CqeType> Op<T, CqeType> {
     /// Create a new operation
-    fn new(data: T, inner: &mut driver::Driver) -> Self {
+    fn new(data: T, index: usize) -> Self {
         Op {
-            index: inner.ops.insert(),
+            index,
             data: Some(data),
             _cqe_type: PhantomData,
             _phantom: PhantomData,
         }
     }
 
-    /// Submit an operation to uring.
+    /// Submit an operation to the driver.
     ///
-    /// `state` is stored during the operation tracking any state submitted to
+    /// `data` is stored during the operation tracking any state for submission to
     /// the kernel.
-    pub(super) fn submit_with<F>(data: T, f: F) -> io::Result<Self>
+    pub(super) fn submit_with<F>(mut data: T, f: F) -> Self
```

Comment:
Submission to the driver is infallible. This now places the seq for submission into the Pending state